### ISSUE TITLE
matplotlib axes.color_cycle deprecated

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#4644]
Building sphinx, there are numerous warnings that

```
/usr/lib/python2.7/dist-packages/matplotlib/__init__.py:874: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
```

I think this arises from https://github.com/astropy/astropy/blob/master/astropy/visualization/mpl_style.py#L64, but am not sure whether we can issues with older versions of matplotlib if we update.

**Comment by bsipocz** [comment]
The same warning came up in gammapy, but somehow I didn't manage to get rid of if even when changed to use ``axes.prop_cycler`` and cycler.

**Comment by astrofrog** [comment]
If we do find a way to make the warnings disappear, can't we conditionally change the ``astropy_mpl_style_1 `` dictionary depending on the MPL version?

**Comment by bsipocz** [comment]
@astrofrog - That sounds like a good plan. Also I vaguely recall that there were some plans to somehow have a versioning for the style file? Maybe that version could also be tied to the mpl version in addition to astropy?

**Comment by stbnps** [comment]
I'm not getting the warning when using the prop_cycle + cycler approach:
`
'axes.prop_cycle': cycler('color',['#348ABD',   # blue
`
@bsipocz have you tried to do it like that?

**Comment by bsipocz** [comment]
@stbnps - I didn't try it in astropy, only in gammapy where it didn't get rid of the warning (in addition to requiring a new dependency, and backward incompatibility etc).

**Comment by stbnps** [comment]
I've just created a pull request: #4666 
The cycler module wasn't being used by astropy, so you may want to include it only
`
if matplotlib.__version__ >= 1.5:
`

**Comment by mhvk** [comment]
Closing since fixed by #5569
